src/writers/dir_builder/mod.rs

```diff
@@ -1,38 +1,40 @@
 // use crate::readers::assembler::WebAPI;
+use std::error::Error;
 use std::fs::{remove_dir_all, DirBuilder, File};
 use std::io::{Result, Write};
 use std::path::{Path, PathBuf};
 
-/// Highest level function abstracts away the struct and all of its calls.
+/// Serves as a flag indicator for the (very limited) types of
+/// sub-directories possible.
 ///
-/// Should succeed at creating the entirety of the directory tree needed.
-pub fn build(base_path_str: String) -> Result<()> {
-    let mut directory_builder = DirectoryBuilder::new(base_path_str);
-    directory_builder.create_base_dir()?;
-
-    // assemble vec of sub-dirs
-    let sub_dir_strs = ["database", "config", "routes", "util", "models"];
-    let mut sub_dirs = Vec::new();
-    for dir_str in &sub_dir_strs {
-        let dir_string = dir_str.to_string();
-        sub_dirs.push(SubDir::new(dir_string));
-    }
-
-    // make the sub-dirs
-    directory_builder.create_sub_directories(&sub_dirs)?;
-
-    Ok(())
-}
-
-/// Very basic sub-directory struct to make it easy and simple to add
-/// sub-directories in the future.
-struct SubDir {
-    path_name: String,
+/// This helps a lot when placing the files in their respective directories (cleanly).
+#[derive(Eq, PartialEq, Hash, Clone)]
+pub enum SubDir {
+    Util,
+    Routes,
+    Models,
+    Other,
 }
 
 impl SubDir {
-    fn new(path_name: String) -> SubDir {
-        SubDir { path_name }
+    /// Matches the incoming path string to an unwrappable enum
+    fn from_path_str(path_str: &String) -> Self {
+        match path_str.as_str() {
+            "routes" => Self::Routes,
+            "util" => Self::Util,
+            "models" => Self::Models,
+            _ => Self::Other,
+        }
+    }
+
+    pub fn as_path_str(&self) -> String {
+        match self {
+            Self::Routes => "routes",
+            Self::Util => "util",
+            Self::Models => "models",
+            Self::Other => "",
+        }
+        .to_string()
```

Comment:
Uncatched enum match from string input should throw error (also change method sig. to `Result`)